Spigot-API-Patches/0221-Create-HoverEvent-from-ItemStack-Entity.patch

```diff
@@ -0,0 +1,130 @@
++    @NotNull
++    public static net.md_5.bungee.api.chat.hover.content.Content hoverContentOf(@NotNull ItemStack itemStack) {
++        return server.hoverContentOf(itemStack);
++    }
++
++    /**
++     * Creates a {@link net.md_5.bungee.api.chat.hover.content.Content} of that {@link Entity} for displaying.
++     *
++     * @param entity     Entity to create the HoverEvent for
++     * @param customName a custom name that should be displayed, if not passed entity name will be displayed
++     * @return the {@link net.md_5.bungee.api.chat.hover.content.Content} of that {@link Entity}
++     */
++    @NotNull
++    public static net.md_5.bungee.api.chat.hover.content.Content hoverContentOf(@NotNull Entity entity, @Nullable String customName) {
++        return server.hoverContentOf(entity, customName);
++    }
++
++    /**
++     * Creates a {@link net.md_5.bungee.api.chat.hover.content.Content} of that {@link Entity} for displaying.
++     *
++     * @param entity     Entity to create the HoverEvent for
++     * @param customName a custom name that should be displayed, if not passed entity name will be displayed
++     * @return the {@link net.md_5.bungee.api.chat.hover.content.Content} of that {@link Entity}
++     */
++    @NotNull
++    public static net.md_5.bungee.api.chat.hover.content.Content hoverContentOf(@NotNull Entity entity, @Nullable BaseComponent customName) {
++        return server.hoverContentOf(entity, customName);
++    }
++
++    /**
++     * Creates a {@link net.md_5.bungee.api.chat.hover.content.Content} of that {@link Entity} for displaying.
++     *
++     * @param entity     Entity to create the HoverEvent for
++     * @param customName a custom name that should be displayed, if not passed entity name will be displayed
++     * @return the {@link net.md_5.bungee.api.chat.hover.content.Content} of that {@link Entity}
++     */
++    @NotNull
++    public static net.md_5.bungee.api.chat.hover.content.Content hoverContentOf(@NotNull Entity entity, @NotNull BaseComponent[] customName) {
++        return server.hoverContentOf(entity, customName);
++    }
+     // Paper end
+ 
+     @NotNull
+diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
+index 7c0a788900c93c29d14d8c45ac5ae3317cf4a94e..91569ac1b25128f74119055569f53d34b16230e8 100644
+--- a/src/main/java/org/bukkit/Server.java
++++ b/src/main/java/org/bukkit/Server.java
+@@ -13,6 +13,8 @@ import java.util.Set;
+ import java.util.UUID;
+ import java.util.function.Consumer;
+ import java.util.logging.Logger;
++import net.md_5.bungee.api.chat.BaseComponent; // Paper
++import net.md_5.bungee.api.chat.hover.content.Content; // Paper
+ import org.bukkit.Warning.WarningState;
+ import org.bukkit.advancement.Advancement;
+ import org.bukkit.block.data.BlockData;
+@@ -1555,5 +1557,44 @@ public interface Server extends PluginMessageRecipient {
+      */
+     @NotNull
+     com.destroystokyo.paper.entity.ai.MobGoals getMobGoals();
++
```

Comment:
These methods belong on ItemFactory, no reason to pollute the root API with these.

Reply:
done